### ISSUE TITLE
Fix all broken links in the new documentation

### ISSUE THREAD

**Issue by evanw555** [linkedin/rest.li#131]
I crawled the new documentation at [rest.li](https://rest.li) looking for broken links and I found the following...

TL;DR: There are 44 broken links in the new documentation and you can see where they are below.

**Page https://linkedin.github.io/rest.li/contribute/howto**
Total links: **16**

**Page https://linkedin.github.io/rest.li/Guice-Dependency-Injection**
Total links: **31**

**Page https://linkedin.github.io/rest.li/Spring-Dependency-Injection**
Total links: **31**

**Page https://linkedin.github.io/rest.li/Rest_li-with-Servlet-Containers**
Link **404**: _"Netty"_ -> https://linkedin.github.io/rest.li/Rest.li-with-Netty
Total links: **33**

**Page https://linkedin.github.io/rest.li/Rest_li-with-Netty**
Total links: **31**

**Page https://linkedin.github.io/rest.li/Request-Response-Framework**
Total links: **27**

**Page https://linkedin.github.io/rest.li/Request-Response-API-(R2)**
Total links: **31**

**Page https://linkedin.github.io/rest.li/D2-Zookeeper-Properties**
Total links: **34**

**Page https://linkedin.github.io/rest.li/Dynamic_Discovery**
Link **404**: _"What is D2"_ -> https://linkedin.github.io/rest.li/Dynamic-Discovery#what-is-d2
Link **404**: _"Terminology"_ -> https://linkedin.github.io/rest.li/Dynamic-Discovery#terminology
Link **404**: _"D2 and Zookeeper"_ -> https://linkedin.github.io/rest.li/Dynamic-Discovery#d2-and-zookeeper
Link **404**: _"D2 Load Balancer"_ -> https://linkedin.github.io/rest.li/Dynamic-Discovery#d2-load-balancer
Link **404**: _"Stores"_ -> https://linkedin.github.io/rest.li/Dynamic-Discovery#stores
Link **404**: _"Operations"_ -> https://linkedin.github.io/rest.li/Dynamic-Discovery#operations
Link **404**: _"Dynamic Discovery Quickstart"_ -> https://linkedin.github.io/rest.li/D2-Quickstart-Tutorial
Total links: **35**

**Page https://linkedin.github.io/rest.li/testsuite_troubleshooting**
Total links: **43**

**Page https://linkedin.github.io/rest.li/Data-FAQ**
Total links: **44**

**Page https://linkedin.github.io/rest.li/howto_add_new_language**
Total links: **43**

**Page https://linkedin.github.io/rest.li/testsuite_how_to**
Total links: **43**

**Page https://linkedin.github.io/rest.li/Rest_li-2_x-upgrade-instructions**
Link **404**: _"Introduction"_ -> https://linkedin.github.io/rest.li/Rest.li-2.x-upgrade-instructions#introduction
Link **404**: _"Source Code Changes"_ -> https://linkedin.github.io/rest.li/Rest.li-2.x-upgrade-instructions#source-code-changes
Link **404**: _"Protocol Changes"_ -> https://linkedin.github.io/rest.li/Rest.li-2.x-upgrade-instructions#protocol-changes
Link **404**: _"Upgrade and Deployment Ordering"_ -> https://linkedin.github.io/rest.li/Rest.li-2.x-upgrade-instructions#upgrade-and-deployment-ordering
Link **404**: _"user guide"_ -> https://linkedin.github.io/rest.li/Rest.li-User-Guide
Link **404**: _"protocol documentation"_ -> https://linkedin.github.io/rest.li/Rest.li-Protocol
Link **404**: _"this page"_ -> https://linkedin.github.io/rest.li/Rest.li-Protocol
Total links: **50**

**Page https://linkedin.github.io/rest.li/Compression**
Total links: **45**

**Page https://linkedin.github.io/rest.li/Rest_li_Avro_conversions**
Link **404**: _"Gradle generateAvroSchema Task"_ -> https://linkedin.github.io/rest.li/Gradle-build-integration#wiki-generateavroschema
Total links: **46**

**Page https://linkedin.github.io/rest.li/How-to-use-projections-in-Java**
Total links: **49**

**Page https://linkedin.github.io/rest.li/Send-Rest_li-Request-Query-In-Request-Body**
Link **404**: _"Rest.li protocol"_ -> https://linkedin.github.io/rest.li/Rest.li-Protocol
Total links: **50**

**Page https://linkedin.github.io/rest.li/Writing-unit-tests-for-Rest_li-clients-and-servers**
Link **404**: _"documentation on Dependency Injection"_ -> https://linkedin.github.io/rest.li/Rest.li-User-Guide#dependency-injection
Total links: **47**

**Page https://linkedin.github.io/rest.li/Scala-Integration**
Link **404**: _"Maven Central"_ -> https://search.maven.org/search?q=g:com.linkedin.sbt-restli%20AND%20a:restli-tools-scala_*
Total links: **48**

**Page https://linkedin.github.io/rest.li/test_suite**
Total links: **56**

**Page https://linkedin.github.io/rest.li/multi_language_compatibility_matrix**
Total links: **48**

**Page https://linkedin.github.io/rest.li/Attachment-Streaming**
Total links: **60**

**Page https://linkedin.github.io/rest.li/Validation-in-Rest_li**
Link **404**: _"Rest.li Filters - Configuration"_ -> https://linkedin.github.io/rest.li/Rest.li-Filters#configuring-filters
Link **404**: _"SpecifyingValidationRules"_ -> https://linkedin.github.io/rest.li/Validation-in-Rest.li#specifying-validation-rules
Total links: **59**

**Page https://linkedin.github.io/rest.li/Projections**
Total links: **48**

**Page https://linkedin.github.io/rest.li/Rest_li-Filters**
Total links: **49**

**Page https://linkedin.github.io/rest.li/modeling/compatibiltiy_check**
Link **404**: _"gradle rest.model.compatibility flag"_ -> https://linkedin.github.io/rest.li/modeling/Gradle-build-integration#compatibility
Total links: **53**

**Page https://linkedin.github.io/rest.li/setup/gradle**
Total links: **30**

**Page https://linkedin.github.io/rest.li/setup/building**
Total links: **26**

**Page https://linkedin.github.io/rest.li/start/unstructured**
Total links: **28**

**Page https://linkedin.github.io/rest.li/start/d2_quick_start**
Link **404**: _"D2 ZookeeperProperties"_ -> https://linkedin.github.io/rest.li/start/D2-Zookeeper-Properties
Total links: **32**

**Page https://linkedin.github.io/rest.li/start/step_by_step**
Link **404**: _"Pegasus DataSchema"_ -> https://linkedin.github.io/rest.li/start/DATA-Data-Schema-and-Templates
Link **404**: _"Data Schema andTemplates"_ -> https://linkedin.github.io/rest.li/start/DATA-Data-Schema-and-Templates
Link **404**: _"Rest.li withServletContainers"_ -> https://linkedin.github.io/rest.li/start/Rest.li-with-Servlet-Containers
Total links: **45**

**Page https://linkedin.github.io/rest.li/modeling/modeling**
Total links: **49**

**Page https://linkedin.github.io/rest.li/DATA-Data-Schema-and-Templates**
Total links: **62**

**Page https://linkedin.github.io/rest.li/Asynchronous-Servers-and-Clients-in-Rest_li**
Link **404**: _"Callback example"_ -> https://linkedin.github.io/rest.li/Asynchronous-Servers-and-Clients-in-Rest.li#using-callbacks-1
Total links: **55**

**Page https://linkedin.github.io/rest.li/Components**
Total links: **47**

**Page https://linkedin.github.io/rest.li/user_guide/unstructured_data**
Total links: **62**

**Page https://linkedin.github.io/rest.li/user_guide/restli_client**
Link **404**: _"designdocuments"_ -> https://linkedin.github.io/rest.li/user_guide/Rest.li-.restspec.json-Format
Total links: **61**

**Page https://linkedin.github.io/rest.li/user_guide/restli_server**
Link **404**: _"Collection Resource Pattern"_ -> https://linkedin.github.io/rest.li/modeling/modeling/#collection
Link **404**: _"Association ResourcePattern"_ -> https://linkedin.github.io/rest.li/user_guide/Modeling-Resources-with-Rest.li#wiki-Association
Total links: **106**

**Page https://linkedin.github.io/rest.li/spec/return_entity**
Total links: **38**

**Page https://linkedin.github.io/rest.li/Rest_li-2_0-response-API**
Total links: **27**

**Page https://linkedin.github.io/rest.li/spec/restspec_format**
Total links: **30**

**Page https://linkedin.github.io/rest.li/spec/protocol**
Link **404**: _"Rest.li 2.0 protocol object URLrepresentation"_ -> https://linkedin.github.io/rest.li/spec/Rest.li-Protocol#url-representation
Link **404**: _"Rest.li 2.0 protocol array URLrepresentation"_ -> https://linkedin.github.io/rest.li/spec/Rest.li-Protocol#url-representation-1
Link **404**: _"Rest.li 2.0 protocol objectHTTP body and headersrepresentation"_ -> https://linkedin.github.io/rest.li/spec/Rest.li-Protocol#http-body-and-headers-representation
Link **404**: _"Rest.li 2.0 protocolarray HTTP body and headersrepresentation"_ -> https://linkedin.github.io/rest.li/spec/Rest.li-Protocol#http-body-and-headers-representation-1
Link **404**: _"How Data is Serialized forTransport"_ -> https://linkedin.github.io/rest.li/spec/DATA-Data-Schema-and-Templates#how-data-is-serialized-for-transport
Link **404**: _"Rest.li 2.0 protocol object URLrepresentation"_ -> https://linkedin.github.io/rest.li/spec/Rest.li-Protocol#url-representation
Link **404**: _"Rest.li 2.0 protocol object HTTP body and headersrepresentation"_ -> https://linkedin.github.io/rest.li/spec/Rest.li-Protocol#http-body-and-headers-representation
Link **404**: _"Rest.li 2.0 protocol array URLrepresentation"_ -> https://linkedin.github.io/rest.li/spec/Rest.li-Protocol#url-representation-1
Link **404**: _"Rest.li 2.0 protocol array HTTP body and headersrepresentation"_ -> https://linkedin.github.io/rest.li/spec/Rest.li-Protocol#http-body-and-headers-representation-1
Link **404**: _"Associationkeys"_ -> https://linkedin.github.io/rest.li/spec/Rest.li-Protocol#association-keys
Link **404**: _"complexkeys"_ -> https://linkedin.github.io/rest.li/spec/Rest.li-Protocol#complex-types-as-keys-in-protocol-20
Link **404**: _"Rest.li 2.0 protocolobjectnotation"_ -> https://linkedin.github.io/rest.li/spec/Rest.li-Protocol#restli-protocol-20-object-representation
Link **404**: _"protocol 2.0 arraynotation"_ -> https://linkedin.github.io/rest.li/spec/Rest.li-Protocol#restli-20-protocol-array-notation
Link **404**: _"protocol 2.0 arraynotation"_ -> https://linkedin.github.io/rest.li/spec/Rest.li-Protocol#restli-20-protocol-array-notation
Total links: **61**

**Page https://linkedin.github.io/rest.li/user_guide/server_architecture**
Total links: **57**

**Page https://linkedin.github.io/rest.li/get_started/quick_start**
Total links: **49**

**Page https://rest.li**
_NOTE:_ redirected to https://linkedin.github.io/rest.li/
Total links: **16**

**Summary**
Visited **180** URL(s)
Crawled **46** page(s)
Found **44** links with HTTP **404** error(s)

**Comment by evanw555** [comment]
There are still a few links I missed. I will send out another quick RB. Here is the log:

**Page https://linkedin.github.io/rest.li/contribute/howto**
Total links: **16**

**Page https://linkedin.github.io/rest.li/Guice-Dependency-Injection**
Total links: **31**

**Page https://linkedin.github.io/rest.li/Spring-Dependency-Injection**
Total links: **31**

**Page https://linkedin.github.io/rest.li/Rest_li-with-Servlet-Containers**
Link **404**: _"Netty"_ -> https://linkedin.github.io/rest.li/Rest.li-with-Netty
Total links: **33**

**Page https://linkedin.github.io/rest.li/Rest_li-with-Netty**
Total links: **31**

**Page https://linkedin.github.io/rest.li/Request-Response-Framework**
Total links: **27**

**Page https://linkedin.github.io/rest.li/Request-Response-API-(R2)**
Total links: **31**

**Page https://linkedin.github.io/rest.li/D2-Zookeeper-Properties**
Total links: **34**

**Page https://linkedin.github.io/rest.li/Dynamic_Discovery**
Link **404**: _"Dynamic Discovery Quickstart"_ -> https://linkedin.github.io/rest.li/D2-Quickstart-Tutorial
Total links: **35**

**Page https://linkedin.github.io/rest.li/testsuite_troubleshooting**
Total links: **43**

**Page https://linkedin.github.io/rest.li/Data-FAQ**
Total links: **44**

**Page https://linkedin.github.io/rest.li/howto_add_new_language**
Total links: **43**

**Page https://linkedin.github.io/rest.li/testsuite_how_to**
Total links: **43**

**Page https://linkedin.github.io/rest.li/Rest_li-2_x-upgrade-instructions**
Total links: **50**

**Page https://linkedin.github.io/rest.li/Compression**
Total links: **45**

**Page https://linkedin.github.io/rest.li/setup/building**
Total links: **26**

**Page https://linkedin.github.io/rest.li/start/unstructured**
Total links: **28**

**Page https://linkedin.github.io/rest.li/start/d2_quick_start**
Total links: **32**

**Page https://linkedin.github.io/rest.li/start/step_by_step**
Total links: **45**

**Page https://linkedin.github.io/rest.li/setup/gradle**
Total links: **30**

**Page https://linkedin.github.io/rest.li/Rest_li_Avro_conversions**
Total links: **46**

**Page https://linkedin.github.io/rest.li/How-to-use-projections-in-Java**
Total links: **49**

**Page https://linkedin.github.io/rest.li/Send-Rest_li-Request-Query-In-Request-Body**
Total links: **50**

**Page https://linkedin.github.io/rest.li/Writing-unit-tests-for-Rest_li-clients-and-servers**
Total links: **47**

**Page https://linkedin.github.io/rest.li/Scala-Integration**
Link **404**: _"Maven Central"_ -> https://search.maven.org/search?q=g:com.linkedin.sbt-restli%20AND%20a:restli-tools-scala_*
Total links: **48**

**Page https://linkedin.github.io/rest.li/test_suite**
Total links: **56**

**Page https://linkedin.github.io/rest.li/multi_language_compatibility_matrix**
Total links: **48**

**Page https://linkedin.github.io/rest.li/Attachment-Streaming**
Total links: **60**

**Page https://linkedin.github.io/rest.li/Validation-in-Rest_li**
Total links: **59**

**Page https://linkedin.github.io/rest.li/Projections**
Total links: **48**

**Page https://linkedin.github.io/rest.li/Rest_li-Filters**
Total links: **49**

**Page https://linkedin.github.io/rest.li/modeling/compatibility_check**
Total links: **53**

**Page https://linkedin.github.io/rest.li/modeling/modeling**
Total links: **49**

**Page https://linkedin.github.io/rest.li/DATA-Data-Schema-and-Templates**
Total links: **62**

**Page https://linkedin.github.io/rest.li/Asynchronous-Servers-and-Clients-in-Rest_li**
Total links: **55**

**Page https://linkedin.github.io/rest.li/Components**
Total links: **47**

**Page https://linkedin.github.io/rest.li/user_guide/unstructured_data**
Total links: **62**

**Page https://linkedin.github.io/rest.li/user_guide/restli_client**
Total links: **61**

**Page https://linkedin.github.io/rest.li/user_guide/restli_server**
Total links: **106**

**Page https://linkedin.github.io/rest.li/spec/return_entity**
Total links: **38**

**Page https://linkedin.github.io/rest.li/Rest_li-2_0-response-API**
Total links: **27**

**Page https://linkedin.github.io/rest.li/spec/restspec_format**
Total links: **30**

**Page https://linkedin.github.io/rest.li/spec/protocol**
Total links: **61**

**Page https://linkedin.github.io/rest.li/user_guide/server_architecture**
Total links: **57**

**Page https://linkedin.github.io/rest.li/get_started/quick_start**
Total links: **49**

**Page https://rest.li**
_NOTE:_ redirected to https://linkedin.github.io/rest.li/
Total links: **16**

**Summary**
Visited **162** URL(s)
Crawled **46** page(s)
Found **3** links with HTTP **404** error(s)

**Comment by evanw555** [comment]
There is only one 404 left, but that's due to the way Maven's artifact search page works. Thus, all the broken links are fixed:

**Page https://linkedin.github.io/rest.li/Scala-Integration**
Link **404**: _"Maven Central"_ -> https://search.maven.org/search?q=g:com.linkedin.sbt-restli%20AND%20a:restli-tools-scala_*
Total links: **48**

**Page https://rest.li**
_NOTE:_ redirected to https://linkedin.github.io/rest.li/
Total links: **16**

**Summary**
Visited **163** URL(s)
Crawled **47** page(s)
Found **1** links with HTTP **404** error(s)

**Comment by Anmol-Singh-Jaggi** [comment]
https://linkedin.github.io/rest.li/user_guide/Asynchronous-Servers-and-Clients-in-Rest_li.md

This is still broken?
